### PR TITLE
Adds icon smoothing for plating.

### PR DESCRIPTION
**Pull request by AndrewMontagne**
Adds icon smoothing for plating to give the impression it is below regular tiles. This also applies to asteroid and lavaland turfs as they inherit from plating.

image

:cl: AndrewMontagne
imageadd: Added icon smoothing for plating.
/:cl:

### CODE REVIEW

**Comment by SamHPurp** [comment]
That looks sick brah

File: code/game/turfs/simulated/floor/plating.dm [hunk 1]
```diff
@@ -24,18 +25,26 @@
 		to_chat(user, "<span class='notice'>You might be able to build ontop of it with some <i>tiles</i>...</span>")
 
 /turf/open/floor/plating/Initialize()
+
+	if(icon == 'icons/turf/floors/plating.dmi') //Hack, lots of subclasses of plating
```
Comment (AndrewMontagne) [inline code/game/turfs/simulated/floor/plating.dm]: This looks reall awful and hardcoded. It is, but having a fuzzy check for any smoothing-enabled *.dmi would be a more expensive operation.

Comment (FlattestGuitar) [reply code/game/turfs/simulated/floor/plating.dm]: Why not just assign a variable to all turfs that use the icon? I'm confused by this.

**Comment by FlattestGuitar** [comment]
You add new asteroid icons, but don't remove the old ones. Are those still used?

File: code/game/turfs/simulated/floor/plating.dm (hunk 1 above)
Comment (AndrewMontagne) [reply code/game/turfs/simulated/floor/plating.dm]: Because that means setting a snowflake var on all subclasses.

**Comment by Ralta** [comment]
Wow, that...actually looks really good!

File: code/game/turfs/simulated/floor/plating.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/game/turfs/simulated/floor/plating.dm]: WHAT THE FUCK

Comment (Iamgoofball) [reply code/game/turfs/simulated/floor/plating.dm]: SWEET JESUS

WHAT IS THAT

**WHAT THE FUCK IS THAT**

**Comment by Iamgoofball** [comment]
who allowed this to be merged oh my god

**Comment by optimumtact** [comment]
goof stop shitposting, a good feature can be worth some crappy code as long as someone cleans it up in a few months time when it gets upstreamed to /tg/station

**Comment by Iamgoofball** [comment]
Really the proper fix here is making everything not a subtype of plating

**Pull request merged**